holoviews/operation/datashader.py

```diff
@@ -1378,8 +1378,12 @@ def _process(self, element, key=None):
         # Potentially needs traverse to find element types first?
         all_allowed_kws = set()
         all_supplied_kws = set()
+        non_default_params = {
+            k: v for k, v in self.param.values().items()
+            if not isequal(v, self.param[k].default)
+        }
```

Comment:
Hmm, having trouble figuring out if this is really correct, it seems like since this is a `ParameterizedFunction` and instance values are never overridden then `non_default_params` will always be empty.

Reply:
One test fails when removing rasterize defaults: `test_segments_aggregate_sum_instance`. Which contains `rasterize.instance`. So by removing the default, this does not work:

``` python
import holoviews as hv
import numpy as np
import xarray as xr
from holoviews.operation.datashader import rasterize

hv.extension("bokeh")

da = xr.DataArray(
    np.arange(100.0).reshape(10, 10),
    coords=dict(x=np.arange(10.0), y=np.arange(10.0)),
)
ri = rasterize.instance(x_range=(-1, 10), y_range=(-1, 10))
ri(hv.Image(da))
```

Reply:
I have updated `non_default_params` to `instance_params` to take account for that in https://github.com/holoviz/holoviews/pull/5767/commits/046c2e0e63445f922f68b36ac025707be2318c4b. 

I have left `isequal` in as I feel it is a good util to have, even though it is not used in this PR anymore.